File: tests/fixtures/sample-js-project-negative-old/services/api-gateway/src/services/user.service.ts
import axios from 'axios';
import { formatUser } from '@sample/shared-utils';

const USER_SERVICE_URL = process.env.USER_SERVICE_URL || 'http://localhost:3001';

export class UserService {
  async findAll() {
    const response = await axios.get(`${USER_SERVICE_URL}/users`);
    return response.data.map(formatUser);
  }

  async findById(id: string) {
    const response = await axios.get(`${USER_SERVICE_URL}/users/${id}`);
    return formatUser(response.data);
  }

  async create(data: { name: string; email: string }) {
    const response = await axios.post(`${USER_SERVICE_URL}/users`, data);
    return formatUser(response.data);
  }

  async delete(id: string) {
    await axios.delete(`${USER_SERVICE_URL}/users/${id}`);
  }
}

File: tests/fixtures/sample-python-project-positive/services/api_gateway/services/user_service.py
"""User service client for the API gateway."""
import os
import requests
from shared.utils.formatters import format_user

USER_SERVICE_URL = os.environ.get("USER_SERVICE_URL", "http://localhost:3001")
HTTP_TIMEOUT = 30


class UserService:
    """Client for communicating with the user microservice."""

    def __init__(self) -> None:
        self._base_url = USER_SERVICE_URL

    def find_all(self) -> list:
        """Fetch all users from the user service."""
        response = requests.get(f"{self._base_url}/users", timeout=HTTP_TIMEOUT)
        return [format_user(u) for u in response.json()]

    def find_by_id(self, user_id: str) -> dict | None:
        """Fetch a single user by their ID."""
        response = requests.get(f"{self._base_url}/users/{user_id}", timeout=HTTP_TIMEOUT)
        return format_user(response.json())

    def create(self, data: dict) -> dict:
        """Create a new user via the user service."""
        response = requests.post(f"{self._base_url}/users", json=data, timeout=HTTP_TIMEOUT)
        return format_user(response.json())

    def delete(self, user_id: str) -> None:
        """Delete a user by their ID."""
        requests.delete(f"{self._base_url}/users/{user_id}", timeout=HTTP_TIMEOUT)
